src/hammer-tech/stackup.py

```diff
@@ -194,6 +197,8 @@ def min_spacing_and_max_width_from_pitch(self, pitch: Decimal) -> Tuple[Decimal,
                 spacing = pitch - width
 
         width = pitch - spacing
+        if self.max_width > 0.0 and width > self.max_width:
```

Comment:
I think some python versions complain about comparing floats (`0.0`) and decimals, but if this passes CI I'm cool with it.